Feature/user refactoring1

jinyongyun opened the pull request:
User 초기 세팅 제작

jinyongyun committed:
#User-application.properties 수정

hibernate 추가, mysql 관련 설정 추가

jinyongyun committed:
#User - User 도메인 추가

유저 엔티티 작성( default 값 향후 추가 가능성 있음)

jinyongyun committed:
#User - UserRepository 작성

유저 레포지터리 작성, 아직 간단한 함수만 정의

jinyongyun committed:
#User - UserService 제작

User관련 CRUD service 메서드 작성

jinyongyun committed:
#User - UserServiceTest 작성

user 서비스에 대한 테스트 작성
 - 유저 임시 생성 메서드
 - 유저 id 검색 메서드
 - 전체 유저 리스트하는 메서드
 - 유저 수정하는 메서드
 - 유저 삭제하는 메서드

테스트 완료, 모두 통과

jinyongyun committed:
#User - UserController 작성

user에 대한 모든 API 작성 예정

jinyongyun committed:
Rename: User 엔티티 제작

 - attribute 1차 수정
 - hash&equals 를 어노테이션으로 변경

jinyongyun committed:
Rename: application properties 를 yaml로 수정

 - jpa/generate-ddl : false 추가
 - jpa/database : mysql 추가

jinyongyun committed:
Feat: UserController frame 제작

API 추가할 예정

jinyongyun committed:
Feat: UserService와 테스트 제작

 - 바뀐 attribute 이름에 맞춰 메서드 변경
 - 컨벤션 참고하여 수정

jinyongyun committed:
Feat: UserRepository 제작

 - 메서드 인자 wrapper로 변경

jinyongyun committed:
Feat: build.gradle과 application.yaml 수정

jinyongyun committed:
Refactor: is_admin 타입 Boolean으로 수정

Godbell commented on src/main/java/community/gdsc/wanted/domain/User.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.Objects;
+
+@Entity
+@Table(name="User")
```
테이블명은 소문자로 작성해 주세요!
```java
@Table(name="user")
```

jinyongyun replied:
수정 완료 👍

Godbell commented on src/test/java/community/gdsc/wanted/service/UserServiceTest.java:
```diff
@@ -0,0 +1,77 @@
+           user.setCoin(10000);
+           user.setIsAdmin(true);
+           //user.setCreated_at();는 생성시 자동 설정
+           user.setIsDeleted(0);
```
`0`보다는 `true`가 적절해 보이네요!
작동하는 데는 문제 없지만, boolean type 값을 넣는 게 테스트 코드만 봐도 isAdmin이 어떤 Column인지 알기 더 쉽겠죠?

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/repository/UserRepository.java:
```diff
@@ -0,0 +1,17 @@
+public interface UserRepository extends JpaRepository<User, Integer> {
+    User findById(int id);
+    void deleteById(Integer id);
+    List<User> findByCoin(Integer coin);
```
`findByCoin()`은 어떤 역할을 하는 메서드인가요?
댓글로 알려 주세요!

jinyongyun replied:
아 의미 없이 넣어봤습니다. 혹시 필요할 지도 몰라서...본격적인 API 작업 후에 계속 업데이트 될 예정입니다!

Godbell replied:
미리 만들기보단 필요할 때 만들기로 하죠!
우선 현재 사용하지 않는 메서드는 삭제해 주세요 :)

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,67 @@
+
+
+import java.util.List;
+
```
import끼리 떨어뜨릴 필요 없습니다 :D

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,67 @@
+
+    //id를 이용해 user를 삭제하는 메서드
+    public void removeUserById(int id){
+        User user = userRepository.findById(id); //아니 orElse가 왜 안붙어
```
코드 품질을 위해 설명하는 것 외의 주석은 삭제해 주세요!

jinyongyun replied:
수정완료!

Godbell commented on src/test/java/community/gdsc/wanted/service/UserServiceTest.java:
```diff
@@ -0,0 +1,77 @@
+
+
+   @Test
+   public void getListUsers(){
```
`getUsers()`라는 이름이 좀 더 간결해 보이네요!

jinyongyun replied:
수정완료!

Godbell commented on src/test/java/community/gdsc/wanted/service/UserServiceTest.java:
```diff
@@ -0,0 +1,77 @@
+
+   @Test
+   public void removeUser(){
+       int testId = 3;
```
상수네요! 다음과 같이 사용해 봅시다.
```java
final int TEST_ID = 3;
```

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/domain/User.java:
```diff
@@ -0,0 +1,64 @@
+public class User {
+    @Id
+    @Column(name = "id")
+    private Integer id; //아이디
```
id는 auto increment 방식으로 생성하는 게 좋습니다.
```java
@GeneratedColumn(strategy = GenerationType.IDENTITY)
```
위 어노테이션을 달아 주면 auto increment column이 되므로 생성자에 값을 넣어 줄 필요가 없습니다.

Godbell reviewed (commented):
1. 두 줄씩 띄워진 코드는 한 줄만 띄워 주세요.
2. brace의 시작과 끝은 줄을 띄우지 말고 붙여 주세요. 함수, 클래스 등 모두 동일합니다.
from:
```java
if (condition) {
    
    statement;

}
```
to:
```java
if (condition) {
    statement;
}
```

jinyongyun committed:
Rename: user 엔티티 클래스 이름 수정

jinyongyun committed:
Feat: yaml 파일에 ddl 방식 update로 수정

jinyongyun committed:
Rename: getListUsers의 이름 getUsers 수정, 공백 제거

jinyongyun committed:
Feat: 임시 예외 클래스 및 Response 객체 클래스

예외처리

jinyongyun committed:
Feat: 암호화에 쓰이는 클래스

AES128 알고리즘 사용 (복호화, 암호화 가능)

jinyongyun committed:
Feat: SignupRequest로 바꾸는 toDto 메서드 작성

jinyongyun committed:
Feat: 회원가입 시 받을 객체 (dto 역할)

 - validation 사용

jinyongyun committed:
Feat: 유저 서비스에 회원가입 API 작성

jinyongyun committed:
Feat: 계속 테이블 업데이트 된 거라 테스트 작동 실패할 수도 있다

jinyongyun committed:
Feat: 유저 아이디 중복 체크를 위해 UserRepository에 관련 함수 추가

jinyongyun committed:
Feat: gradle에 형식 체크를 위한 validation 추가

Godbell commented on src/main/java/community/gdsc/wanted/config/secret/Secret.java:
Secret Key를 Git에 올라가는 파일에 관리하는 것은 위험합니다.
그리고 이 부분은 차후 Auth 개발하면서 따로 작성할 부분이라 코드에서 제외 바랍니다.
비밀번호 암호화 저장이나 시크릿 키 참조와 같은 기능은 우선 모두 제외해 주세요!

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/domain/User.java:
```diff
@@ -0,0 +1,68 @@
+    @Column(name = "coin")
+    private Integer coin;
+
+    @Column(length = 45, name = "is_admin")
```
length 값 제외 바랍니다!

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,113 @@
+    UserRepository userRepository;
+
+    //회원가입
+    public SignupRequest createUser(SignupRequest postUserReq) throws BaseException {
```
signup 부분은 제쪽에서 개발할게요
Spring Security 사용할 부분이라 다르게 개발해야 합니다.
우선 해당 메서드 및 연관 메서드는 모두 제외해 주세요.

jinyongyun replied:
이거 그냥 베이스만 남겨놓았습니다!

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/utils/AES128.java:
이 부분은 타 인코더 사용하여 구현할 예정입니다.
Auth 및 Secret 관련 내용은 제외해 주세요.

jinyongyun replied:
수정완료!

Godbell commented on src/test/java/community/gdsc/wanted/service/UserServiceTest.java:
1. 사용하지 않는 주석은 제거해 주세요. 필요했던 내용은 이전 커밋에서 확인 바랍니다.
2. 자동 포맷 옵션 킨 채로 저장해서 형식 한 번 맞춰 주세요 :)

jinyongyun replied:
저 왜 이거 켜 놓았는데 적용 안되는 것 같죠

jinyongyun replied:
수정완료!

Godbell reviewed (commented):
- 자동 포맷팅 옵션 설정하시고 각 파일 한 번씩 저장해서 적용해 주세요!
- 가입, 로그인, 로그아웃, 비밀번호 인코딩 부분은 Spring Security 붙이면서 해결하겠습니다. 해당 부분은 제외해 주세요.

jinyongyun committed:
Remove: Secutiry 관련 파일 삭제

jinyongyun committed:
Feat: 회원 가입 API 베이스 로직만 작성

 @godbell 수정은 맡기겠습니다! 이 뒤로는 아무도 못지나간다!

jinyongyun committed:
Refactor: findByCoint 삭제

Godbell commented on src/main/java/community/gdsc/wanted/RequestModel/SignupRequest.java:
```diff
@@ -0,0 +1,67 @@
+    private String username;
+    @NotEmpty(message = "비밀번호는 필수 입력값입니다.")
+    @Size(min=6, max=40)
+    @Pattern(regexp = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[~!@#$%^&*()+|=])[A-Za-z\\d~!@#$%^&*()+|=]{8,16}$", message = "비밀번호는 8~16자 영문 대 소문자, 숫자, 특수문자를 사용하세요.")
```
비밀번호는 AI 계획서에 8~40자 모든 출력 가능 문자 허용으로 정의했습니다.
입력 가능한 문자는 모두 가능이므로 `Pattern` 어노테이션에서 `message` 부분은 없애도 될 것 같네요.
길이는 8~40, 정규식은 다음과 같이 정의해 주세요.

```java
@Pattern(regexp = "^[ -~]+$")
```

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/RequestModel/SignupRequest.java:
```diff
@@ -0,0 +1,67 @@
+    @NotEmpty(message = "비밀번호 확인은 필수 입력값입니다.")
+    @Size(min=6, max=40)
+    @Pattern(regexp = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[~!@#$%^&*()+|=])[A-Za-z\\d~!@#$%^&*()+|=]{8,16}$", message = "비밀번호는 8~16자 영문 대 소문자, 숫자, 특수문자를 사용하세요.")
+    private String confirmPassword;
```
위에 언급한 비밀번호 형식에 맞춰 주세요!
이름은 `passwordConfirm`이 좋아 보입니다.
동사 시작 이름은 함수에만 사용해 주세요!

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/RequestModel/SignupRequest.java:
```diff
@@ -0,0 +1,67 @@
+    private String regionDepth2;
+
+    @NotEmpty(message = "지역은 필수 입력값입니다.")
+    private String regionDepth3;
```
각각의 depth에 대해서 안내 메시지에 '광역시/도', '시/군/구', '읍/면/동'을 기입해 주세요.

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/controller/UserController.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    @ResponseBody
+    @PostMapping("/")    // POST 방식의 요청을 매핑하기 위한 어노테이션
+    public BaseResponse<SignupRequest> createUser(@Valid @RequestBody SignupRequest postUserReq) {
```
이거 일단 요대로 두고 나중에 제가 예외처리 모듈 만들고 나서 그걸로 고치겠습니다.

jinyongyun replied:
👍🏻

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,103 @@
+            System.out.println("userId?? "+username);
+            if(chkID.getUsername().equals(username)){
+                System.out.println("이미 데베에 존재해~!!");
+                duplicate =  true;
```
Line 41에서 바로 `return true`해줬듯, `duplicate` 변수 삭제하고 바로 `return false`해주는 게 더 통일성 있을 것 같아요!

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    // 유저 작성하는 메서드
+    public User writeUser(User user){
```
`addUser`쪽이 좀 더 자연스러워 보입니다.

jinyongyun replied:
수정완료!

Godbell commented on src/main/java/community/gdsc/wanted/service/UserService.java:
```diff
@@ -0,0 +1,103 @@
+    //id를 이용해 user를 삭제하는 메서드
+    public void removeUserById(int id){
+        User user = userRepository.findById(id);
+        if(user!=null)
```
Early return으로 구현해 주세요!
```java
if (user == null) return;
userRepository.deleteById(id);
```
예외도 던져야 할 것 같긴 한데 이 부분은 제가 나중에 예외처리 모듈 만들고 추가할게요.
아래 주석만 하나 달아주세요 :D
```java
// TODO: 예외 throw
```

jinyongyun replied:
👍🏻

Godbell reviewed (commented):
늦은 리뷰 죄송합니다ㅠㅠ
이번 수정점 반영되면 merge 가능할 것 같네요!

jinyongyun committed:
Feat: 수정사항 tiny

 - userController에서 getPasswordConfirm으로 이름 변경
 - SignupRequest 비밀번호 패턴 변경
 - UserService에서 addUser로 메서드 이름 변경

jinyongyun committed:
Merge branch 'develop' into feature/User-refactoring1

Godbell committed:
Update SignupRequest.java

Godbell committed:
Chore: appplication.yaml 조정

Godbell reviewed (approved):
LGTM :)
